[2024-07-14T00:20:43.650+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-07-14T00:20:43.836+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: from_csv_to_pgsql.save_file_md_ledger_account_s.csv scheduled__2024-07-03T00:00:00+00:00 [queued]>
[2024-07-14T00:20:43.868+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: from_csv_to_pgsql.save_file_md_ledger_account_s.csv scheduled__2024-07-03T00:00:00+00:00 [queued]>
[2024-07-14T00:20:43.870+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 2
[2024-07-14T00:20:43.919+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): save_file_md_ledger_account_s.csv> on 2024-07-03 00:00:00+00:00
[2024-07-14T00:20:43.966+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'from_csv_to_pgsql', 'save_file_md_ledger_account_s.csv', 'scheduled__2024-07-03T00:00:00+00:00', '--job-id', '122', '--raw', '--subdir', 'DAGS_FOLDER/from_csv_to_pgsql.py', '--cfg-path', '/tmp/tmpm22pxoom']
[2024-07-14T00:20:43.982+0000] {standard_task_runner.py:91} INFO - Job 122: Subtask save_file_md_ledger_account_s.csv
[2024-07-14T00:20:44.016+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=1620) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-07-14T00:20:44.018+0000] {standard_task_runner.py:63} INFO - Started process 1646 to run task
[2024-07-14T00:20:44.149+0000] {task_command.py:426} INFO - Running <TaskInstance: from_csv_to_pgsql.save_file_md_ledger_account_s.csv scheduled__2024-07-03T00:00:00+00:00 [running]> on host f4c752a3ffdd
[2024-07-14T00:20:44.485+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='from_csv_to_pgsql' AIRFLOW_CTX_TASK_ID='save_file_md_ledger_account_s.csv' AIRFLOW_CTX_EXECUTION_DATE='2024-07-03T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-07-03T00:00:00+00:00'
[2024-07-14T00:20:44.490+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-07-14T00:20:44.767+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-07-14T00:20:44.771+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/from_csv_to_pgsql.py", line 13, in load_csv_to_spark
    spark = spark_session()
            ^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/from_csv_to_pgsql.py", line 9, in spark_session
    spark = SparkSession.builder.appName("AirflowETL").getOrCreate()
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/sql/session.py", line 497, in getOrCreate
    sc = SparkContext.getOrCreate(sparkConf)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 515, in getOrCreate
    SparkContext(conf=conf or SparkConf())
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 201, in __init__
    SparkContext._ensure_initialized(self, gateway=gateway, conf=conf)
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/context.py", line 436, in _ensure_initialized
    SparkContext._gateway = gateway or launch_gateway(conf)
                                       ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pyspark/java_gateway.py", line 107, in launch_gateway
    raise PySparkRuntimeError(
pyspark.errors.exceptions.base.PySparkRuntimeError: [JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.
[2024-07-14T00:20:44.837+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=from_csv_to_pgsql, task_id=save_file_md_ledger_account_s.csv, run_id=scheduled__2024-07-03T00:00:00+00:00, execution_date=20240703T000000, start_date=20240714T002043, end_date=20240714T002044
[2024-07-14T00:20:44.903+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 122 for task save_file_md_ledger_account_s.csv ([JAVA_GATEWAY_EXITED] Java gateway process exited before sending its port number.; 1646)
[2024-07-14T00:20:44.936+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-07-14T00:20:44.974+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
